CMakeSetCompilerOptions.cmake

```diff
@@ -0,0 +1,35 @@
+# these are always defined
+add_definitions(-DNOMINMAX -D_CRT_SECURE_NO_WARNINGS)
```

Comment:
I suspect NOMINMAX is also needed on UNIX, while D_CRT_SECURE_NO_WARNINGS is totally msvc stuff.. I'll try to compile on linux without nominmax defined to see if something bad happens :-)